### DISCUSSION TITLE
Plataformas de e-commerce

### DISCUSSION THREAD

**Discussion by celsomtrindade** [frontendbr/forum#724] [Perguntas e Respostas]
Fala pessoal, tudo certo?

### Para por em contexto
> Sempre trabalhei com desenvolvimento mas nunca cheguei a trabalhar na área de e-commerce, sempre em sistemas internos de empresas, sites institucionais, etc.. Nunca cheguei a utilizar `wordpress` (faz muito tempo que não trabalho com ele), ultimamente tenho trabalhado com `Angular (2+)`, enfim, de modo mais "bruto" no desenvolvimento (pois a empresa que trabalho já possui uma certa "plataforma" interna de gerenciamento de clientes).

<hr />

Estou iniciando um projeto (paralelo) onde faremos um e-commerce, e é aí que minha dúvida começa.

Andei pesquisando, lendo, vendo vídeos e notei que por quase unanimidade a sugestão é: **utilizar uma plataforma de e-commerce pronta**. O problema é que, dentre as plataformas que encontrei, não consegui ver uma real "vantagem", além de existir alguns pontos que acho que atrapalham um pouco o início, como:

- Mensalidade (alguns casos são valores altos para se iniciar );
- Produtos limitados (alguns com apenas 25, 50);
- Usuários limitados (alguns com apenas 1 ou 2);

Das plataformas que encontrei que não possuíam valores altos, ou essas limitações, alguns pontos não ficaram claros, como: 

- Sem integração de pagamento, ou o desenvolvedor deve implementar isso
- Possuir % sobre vendas
- Utilizar temas fixos, ou apenas layout de bootstrap
- entre outros...

Ou seja, analisando bem as plataformas, eu não consigo ver uma real vantagem em utilizar elas. Consigo ver seu uso em um cenário de e-commerce já bem implementado, onde a empresa possui um faturamento suficiente para arcar com esses custos.

<hr />

Outro ponto que favorece meu olhar "negativo" para o uso de plataformas é devido ao fato de eu estar acostumado com desenvolvimento das ferramentas/funcionalidades que utilizamos. Dentre os projetos que tenho, já trabalhei/trabalho com:

- Gerenciamento de produtos
- Possuímos uma integração de pagamento (pagseguro e cielo) para cobrança dos nossos clientes
- Desenvolvimento frontend bem automatizado
- Gerenciamento de usuários (cadastro, login, login social, recuperação de senha, etc)

Ou seja, eu já me envolvi com algumas funcionalidades que são utilizadas dentro de um e-commerce, porém não tem a característica de venda de um produto para um cliente final.

<hr />

Como disse no início, sou extremamente leigo no ramo de e-commerce propriamente dito, dentre as diversas plataformas que já vi (e não foram poucas), todas me parecem muito semelhantes e dentro dos pontos que coloquei acima.

Por isso estou com essa dificuldade de decidir o que fazer.

Devo utilizar uma plataforma e-commerce? Desenvolver uma própria, sendo que já possuo algumas funcionalidades que necessitam apenas de adaptação? O que vou ganhar ao utilizar uma plataforma? O que vou ganhar ao desenvolver um e-commerce próprio?

Qualquer outro comentário relacionado a isso será muito bem vindo, pois quero me aprofundar e entender melhor o cenário antes de escolher o caminho a seguir, pois depois acho que a mudança de um para outro será pior e mais custosa do que já iniciar no melhor cenário.

**Comment by mksanti** [comment]
Bem, eu n sei que plataformas você estava pesquisando. Mas o Woocommerce, Magento e similares não se encaixam nesses quesitos levantados. Você tem total controle sobre o código, possuem diversas integrações com as mais variadas soluções e são free.

--- Edit ---

Algumas disponibilizam até API para sua aplicação.

**Comment by felipeuntill** [comment]
Recomendo a [VTEX](https://pt.vtex.com/), tem seus problemas assim como qualquer solução, sendo ela proprietária ou não. Mas ao meu ver é disparado a melhor solução para e-commerce hoje.

**Comment by celsomtrindade** [comment]
@mikejavier Woocommerce pelo que vi é com base em Wordpress, o que estou tentando evitar, pois teria que "aprender", em teoria, 2 novos métodos de trabalho (não descarto, mas pretendo ver outras). O Magento estou dando uma lida mais agora, inclusive instalando localmente para testar. Alguma sugestão de quais utilizam API? E como funcionaria isso?

<hr />

@felipeuntill Eu vi bastante coisa sobre a VTEX, porém os custos para iniciar não me pareceram tão atrativos. Pelo menos no site deles a informação é de que há uma taxa mensal em torno de R$700,00 + % sobre faturamento. Acho isso muito inviável para iniciar um e-commerce.

**Comment by weltonsanches** [comment]
Tem também o Prestashop e Opencart. Lembrando que Woocommerce, Magento, Prestashop, Opencart são feitos em PHP é interessante que você conheça a linguagem pra aproveitar melhor.

**Comment by mksanti** [comment]
@celsomtrindade Não entendi a parte de "2 métodos de trabalho" o Woocommerce por ser plugin do WP nada mais do que aumenta suas funcionalidades. Neste caso teria que aprender como usar e desenvolver em WP.  Outra alternativa seria fazer a instalação e alimentar o sistema via API, assim vc fica livre para desenvolver como se sentir confortável. Segue os links.

http://v2.wp-api.org/
https://docs.woocommerce.com/document/woocommerce-rest-api/

**Comment by renanlara** [comment]
@celsomtrindade, o WooCommerce é um plugin do WordPress, é como se ele transformasse o WordPress em ecommerce. Uma vez que você tem um certo conhecimento em php e saiba ler a documentação do WordPress, fazer qualquer coisa no WooCommerce não será trabalho nenhum, pois ele reaproveita por assim dizer todas as funções do WP. Eu já desenvolvi algumas lojas no nele, uma bem simples que está no ar já tem alguns meses é essa aqui https://reservadacanastra.com/loja/. Sinceramente, a facilidade de se implementar um ecommerce com o WooCommerce é sureal, sugiro que faça pelo menos alguns testes.

Essa loja da Reserva da Canastra foi um tema desenvolvido totalmente do zero utilizando o [Odin Framework](https://github.com/wpbrasil/odin).

**Comment by didiraja** [comment]
Eu particularmente tive problemas com o VTEX lá no trabalho, por causa do sistema meio fechado. Digo isso em relação a código. Como todo CMS, ele é trabalhado em templates, que você vai atualizando separadamente, mas achei bastante complicado e se tornando um dificultador no nosso caso, que somos agência de publicidade e precisamos atochar código em tudo que é canto.

Você tá me parecendo procurar liberdade e aproveitar seus conhecimentos de código já existentes, o WooCommerce me parece ser a melhor opção (Não conheço nada de Magento). Talvez te assuste o fato de aprender WP, mas a documentação deles é excelente e o sistema é simples. Acho que ce ganha mais aprendendo Wordpress.

**Comment by brunomosciatti** [comment]
Recomendo a Xtech; estou a mais de 1 ano com eles

**Comment by munizart** [comment]
Não tenho muitas experiencia do outro lado, mas no meu trabalho a gente tem muitos clientes que usam essas plataformas.
O principal ponto que vejo é que as lojas menores não costumam ter logo de plataforma no rodapé, então acredito que ou desenvolveram uma própria ou usaram magento.

Sobre vtex, algumas lojas maiores usam. A maior dificuldade que tive com eles foi algumas questões de CORs por estar num subdomínio diferente (resolvemos usando XDM).
Essas lojas que usam vtex costumam mais ou menos o mesmo formato de navegação, mas layouts bem diferentes, então acho que algumas coisas são mais customizáveis que outras. Uma delas inclusive teve um redesign enorme e passou a usar React.

Minha dica seria achar lojas online com um perfil parecido com teu caso e ver o que eles estão usando
